torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -1204,6 +1185,17 @@ XLAGraphExecutor::TryRunCachedSync(
   TORCH_LAZY_VALUE_METRIC("TensorsGraphSize", po_data->post_order.size());
   TF_VLOG(5) << "TensorsGraphSize=" << po_data->post_order.size();
 
+  if (runtime::sys_util::GetEnvBool("XLA_AUTO_SPMD", false)) {
+    // TODO(yeounoh) we may be able to update the cache to avoid this.
```

Comment:
TODO, we need to in-place update the original XLATensor to point to the new sharded data. So the last cached computation (auto-sharded) and the parameters are not in sync, requiring additional resharding.

Reply:
check with @JackCaoG

Reply:
Added comments.